services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeService.kt

```diff
@@ -70,77 +71,61 @@ class HankeService(
     @Transactional
     fun createHanke(
         request: CreateHankeRequest,
-        founder: HankekayttajaInput? = null,
+        securityContext: SecurityContext,
+        alueet: List<NewHankealue>? = null,
         generated: Boolean = false,
     ): Hanke {
-        val userId = currentUserId()
-
-        val entity: HankeEntity = createEntityFromCreateRequest(request, generated, userId)
+        val entity: HankeEntity =
+            createEntityFromCreateRequest(request, generated, securityContext.userId())
 
-        val loggingEntryHolder = prepareLogging(entity)
-        copyYhteystietosToEntity(request, entity, userId, loggingEntryHolder, mutableMapOf())
+        alueet?.let {
```

Comment:
Creating a new hanke through hanke controller will not have any areas.

Would it make sense to have two different functions, one for creating a regular hanke and one for cable report situations? These of course have common functionalities which could also be in a separate function.

Reply:
I fiddled around a bit with this and split the core of createHanke into two parts, `createNewEntity` and `saveCreatedHanke`. `createHanke` now only does these two steps. `generateHankeFrom` calls those methods and adds the areas in-between. I like that now `generateHankeFrom` doesn't have to create a DTO in the middle of processing, and the transactions are clearer.